### PR TITLE
blocksync: export errors

### PR DESCRIPTION
**Pull request by pysel**
---

Contributes: #1140 
#### Changes

Export errors in `blocksync` package. More context: #1140

### CODE REVIEW

File: blocksync/errors.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+// ErrInvalidBase is returned when peer informs of a status with invalid height
+type ErrInvalidHeight struct {
+	Height int64
+	Reason string
+}
+
+func (e ErrInvalidHeight) Error() string {
+	return fmt.Sprintf("Invalid height %v. %s", e.Height, e.Reason)
```
Comment (thanethomson) [inline blocksync/errors.go:22]:
```suggestion
	return fmt.Sprintf("invalid height %v: %s", e.Height, e.Reason)
```

Errors might be nested.

File: blocksync/errors.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+// ErrInvalidBase is returned when peer informs of a status with invalid base
+type ErrInvalidBase struct {
+	Base   int64
+	Reason string
+}
+
+func (e ErrInvalidBase) Error() string {
+	return fmt.Sprintf("Invalid base %v. %s", e.Base, e.Reason)
```
Comment (thanethomson) [inline blocksync/errors.go:32]:
```suggestion
	return fmt.Sprintf("invalid base %v: %s", e.Base, e.Reason)
```

File: blocksync/errors.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+func (e ErrUnknownMessageType) Error() string {
+	return fmt.Sprintf("unknown message type %T", e.Msg)
+}
+
+type ErrReactorValidation struct {
+	Err error
+}
+
+func (e ErrReactorValidation) Error() string {
+	return fmt.Sprintf("Reactor validation error %v", e.Err)
```
Comment (thanethomson) [inline blocksync/errors.go:48]:
```suggestion
	return fmt.Sprintf("reactor validation error: %v", e.Err)
```

**Review by thanethomson — CHANGES_REQUESTED** [review]
Looking good so far, thanks @pysel! Just a few minor nits.

File: blocksync/errors.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+type ErrUnknownMessageType struct {
+	Msg proto.Message
+}
+
+func (e ErrUnknownMessageType) Error() string {
+	return fmt.Sprintf("unknown message type %T", e.Msg)
+}
+
+type ErrReactorValidation struct {
```
Comment (thanethomson) [inline blocksync/errors.go:43]: Given that this embeds another error, I'd recommend implementing an `Unwrap` method that returns the inner error as per https://pkg.go.dev/errors#Unwrap.

**Comment by pysel** [comment]
hey @thanethomson, comments addressed, I think this PR is good to go!

**Review by thanethomson — APPROVED** [review]
Amazing, thanks @pysel!

For now I'm marking this to be backported to v0.38.x, but if anyone needs it backported further back then let me know. cc @p4u

**Pull request merged**